### PR TITLE
Redux in React Components

### PR DESCRIPTION
**Pull request by Epaltechs**
### In this milestone of the Bookstore project, I:
- Change the default state in the books reducer from an empty array to an array with a few books.
- Import Redux Provider and store in the main component of the React app.
- Connect the React app to the Redux store.
- Make sure that list of books displays books from the Redux store.
- Use useSelector to consume the state.
- Add functionality for the "Add new" button click and Import Redux useDispatch in the Form.
- Add functionality for the "Remove" button click and Import Redux useDispatch in Books.
- Add functionality for the "Check status" button click and Import Redux useDispatch in Categories.

### CODE REVIEW

File: src/components/books/Books.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import { useSelector } from 'react-redux';
+import Book from '../book/Book';
+
+const Books = () => {
+  const books = useSelector((state) => state.books);
+  return (
+    <div className="books">
+      {books.length > 0 ? (
+        books.map(({ id, title, catagory }) => (
+          <Book key={id} id={id} title={title} catagory={catagory} />
```
Comment (ShahierNashaat) [inline src/components/books/Books.js]: - [ ] Kindly remove the category from here as it is undefined and causing an error or you can give it a default value.

File: src/components/book/Book.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import { removeBook } from '../../redux/books/books';
+
+const Book = ({ id, title, catagory }) => {
```
Comment (ShahierNashaat) [inline src/components/book/Book.js]: - [ ] Kindly remove the category from here as it is undefined and causing an error or you can give it a default value.

File: src/components/book/Book.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  const dispatch = useDispatch();
+  return (
+    <div>
+      <h2>{title}</h2>
+      <h5>{catagory}</h5>
```
Comment (ShahierNashaat) [inline src/components/book/Book.js]: - [ ] Kindly remove the category from here as it is undefined and causing an error or you can give it a default value.

File: src/components/categories/Categories.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+const Categories = () => {
+  const dispatch = useDispatch();
+  const showTextUnderConstruction = () => {
+    dispatch({ type: 'ACTIVE' });
```
Comment (ShahierNashaat) [inline src/components/categories/Categories.js]: - [ ] Kindly be noticed that the `'ACTIVE'` isn't the actual value that you should send to the reducer please change it to the actual value which is `'redux/categories/ACTIVE''`.

**Review by ShahierNashaat — CHANGES_REQUESTED** [review]
Hi @Epaltechs , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Epaltechs** [comment]
### As required, I have made the following changes:
- Remove the expression 'category' from Book,js, and Books.js files.
- Replace 'ACTIVE' with 'redux/categories/ACTIVE' in Categories.js file.

**Review by mateo951 — APPROVED** [review]
### Hi @Epaltechs, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🧐 
### Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**